### PR TITLE
Implementation of Interpolating the username through the given data

### PR DESCRIPTION
**Pull request by ippokratis12**
# Description

For starters i created a function in the MasterWindow called find_user_in_conversations. In this function i find the username of the user. This function basically is pretty simple. Firstly i find the conversations in which there are only two participants. After doing that i'm trying to find the same name twice in two different conversations. Doing that i find the name of the user because we already know that the conversations belong to the user. So finding the same name in two different conversations means that this name is the user's username. When we find the username we return it.

Also i needed to change the open_file_explorer function in ConfigurationPage class and in SettingsPopup class in order to take properly the name from function find_user_in_conversation and put the username in the entry box. Basically whenever the user choose the directory either from the configuration page or from the settings it automatically finds the user's username. However when you open the settings you can also put some other name there if the user wants to. But each time you choose the directory the user's username is found automatically.

In the end i also created a deutch.py file

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I tested it with the json files i downloaded from my account in facebook

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Main.py
Comment (Kubis10) [inline Main.py:1]: You deleted changes from task to count average messages by time intervals added in #43

**Review by Coder2Mo — CHANGES_REQUESTED** [review]
well i really like the changes you did it really does well but what if we can add some error handling to the find_user_in_conversation function to handle if we did find the use or not this is an example of what am saying 

```
def find_user_in_conversations(self):
    try:
        # Find all conversation files
        participant_counts = defaultdict(int)
        two_person_chats = []

        # List conversations
        for conversation in glob.glob(f'{self.directory}/*'):
            # Extract participants
            participants = set()

            for file in glob.glob(f'{conversation}/*.json'):
                with open(file, 'r') as f:
                    data = json.load(f)
                    for participant in data.get('participants', []):
                        name = participant.get('name', '').encode('iso-8859-1').decode('utf-8')
                        if name:
                            participants.add(name)

            # Check if the chat has exactly two participants
            if len(participants) == 2:
                two_person_chats.append(participants)

        # Find a common participant
        for participants in two_person_chats:
            for participant in participants:
                participant_counts[participant] += 1
                if participant_counts[participant] > 1:
                    return participant  # Found a common participant

        # No common participant found
        return None
    except Exception as e:
        # Handle exceptions, log, or print an error message
        print(f"Error in find_user_in_conversations: {e}")
        return None
```

for the open_file_explorer function i added some error handling too to ensure that the selected directory of the json data we added is valid and contains the expected files

```
    def open_file_explorer(self):
        try:
            selected_directory = tk.filedialog.askdirectory()

            if selected_directory:
                path = os.path.join(selected_directory, "")  # Ensure the path ends with a slash
                self.update_directory_ui(path)

                # Update the directory in the controller
                self.controller.directory = path

                # Attempt to find the user name based on the new directory
                self.controller.update_username_from_conversations(path)
                user_name = self.controller.username

                if user_name:
                    self.username_label.delete(0, tk.END)
                    self.username_label.insert(0, user_name)
            else:
                # Handle the case where no directory is selected
                self.update_directory_ui(self.module.TITLE_NO_SELECTION)
        except Exception as e:
            # Handle unexpected errors gracefully
            print(f"Error in open_file_explorer: {e}")

    def update_directory_ui(self, path):
        # Update the directory label in the UI
        self.directory_label.config(text=path)
```

that's all i did but you really done a great work my friend

**Comment by ippokratis12** [comment]
do i need to do anything more? Or are you going to merge this function to the main branch?

File: Main.py
Comment (ippokratis12) [reply Main.py:1]: i think it was an accident. Maybe i didn't notice. I'm sorry about it but i didn't intend to do something like that. Do i need to do anything more or is the feature ok by you? If yes will it be merged to the main code?

**Comment by Kubis10** [comment]
Just add this to your fork and commit it, I will merge it

File: Main.py
Comment (Kubis10) [reply Main.py:1]: Just add this what you deleted, and it will be fine, I will test it then

File: langs/Deutch.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+TITLE_CONVERSATION_TYPE = 'Gesprächstyp'
+TITLE_PEOPLE = 'Menschen'
+TITLE_AND_MESSAGES = 'und Nachrichten'
+TITLE_START_DATE = 'Datum der ersten Nachricht'
+TITLE_AVERAGE_MESSAGES = 'Durchschnittliche Nachrichten pro'
+TITLE_PER_DAY = 'Tag'
+TITLE_PER_WEEK = 'Woche'
+TITLE_PER_MONTH = 'Monat'
+TITLE_PER_YEAR = 'Jahr'
+
```
Comment (Kubis10) [inline langs/Deutch.py:42]: Also delete this empty lines

**Review by Kubis10 — APPROVED** [review]

**Comment by Kubis10** [comment]
Great work

**Comment by ippokratis12** [comment]
Now do i close the pull request or do i need to do anything about the merge section? Or are you going to handle the rest?

**Comment by Kubis10** [comment]
I will handle this

**Pull request merged**